plugin-maven/src/main/java/com/diffplug/spotless/maven/groovy/Groovy.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2016 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.maven.groovy;
+
+import java.util.Set;
+
+import com.diffplug.common.collect.ImmutableSet;
+import com.diffplug.spotless.maven.FormatterFactory;
+import com.diffplug.spotless.maven.generic.LicenseHeader;
+
+/**
+ * A {@link FormatterFactory} implementation that corresponds to {@code <java>...</java>} configuration element.
+ * <p>
+ * It defines a formatter for java source files that can execute both language agnostic (e.g. {@link LicenseHeader})
+ * and java-specific (e.g. {@link Eclipse}) steps.
+ */
+public class Groovy extends FormatterFactory {
+
+	private static final Set<String> DEFAULT_INCLUDES = ImmutableSet.of("src/main/groovy/**/*.groovy", "src/test/groovy/**/*.groovy");
+	private static final String LICENSE_HEADER_DELIMITER = "package ";
```

Comment:
Perhaps this pattern makes sense here too:

- https://github.com/diffplug/spotless/blob/b7f8c551a97dcb92cc4b0ee665448da5013b30a3/plugin-gradle/src/main/java/com/diffplug/gradle/spotless/GroovyExtension.java#L59
- https://github.com/diffplug/spotless/blob/b7f8c551a97dcb92cc4b0ee665448da5013b30a3/plugin-gradle/src/main/java/com/diffplug/gradle/spotless/JavaExtension.java#L47-L49